### ISSUE TITLE
OSError: /lib/x86_64-linux-gnu/libm.so.6: version `GLIBC_2.27' not found (required by /home/user/anaconda3/envs/hbk_gcn/lib/python3.9/site-packages/torch_spline_conv/_basis_cuda.so)

### ISSUE THREAD

**Issue by kkwd22** [pyg-team/pytorch_geometric#3477]
## Environment

* OS:ubuntu 16.04
* Python version:3.9
* PyTorch version:1.10
* CUDA/cuDNN version:10.2
* GCC version:
* How did you try to install PyTorch Geometric and its extensions (wheel, source):
* Any other relevant information:

## Checklist

- [ ] I followed the [installation guide](https://pytorch-geometric.readthedocs.io/en/latest/notes/installation.html).
- [ ] I cannot find my error message in the [FAQ](https://pytorch-geometric.readthedocs.io/en/latest/notes/installation.html#id1).
- [ ] I set up CUDA correctly and can compile CUDA code via `nvcc`.
- [ ] I do have multiple CUDA versions on my machine.

**Comment by rusty1s** [comment]
How did you try to install the extension packages and what's your GCC version?

**Comment by hcw0098** [comment]
same error 

- OS : ubuntu 16.04
- python : 3.8/3.9, try them both
- pytorch: 1.10.0
- CUDA 10.2
- GCC version 7.5.0
- PyG isntall : using "conda install pyg -c pyg -c conda-forge"
 import torch_geometric successfully, but when I try to import torch_geometric.nn， this error occurred

**Comment by hcw0098** [comment]
same error 

- OS : ubuntu 16.04
- python : 3.8/3.9, try them both
- pytorch: 1.10.0
- CUDA 10.2
- GCC version 7.5.0
- PyG isntall : using "conda install pyg -c pyg -c conda-forge"
 import torch_geometric successfully, but when I try to import torch_geometric.nn， this error occurred

**Comment by rusty1s** [comment]
Does that help? https://github.com/NVIDIA/TensorRT/issues/1124

**Comment by yuhongyu0721** [comment]
I meet the same problem when using Pytorch 1.10 and pyg with cuda 11.3.
I install with the following commands:
`
conda create -n pasp_gnn pytorch torchvision torchaudio cudatoolkit=11.3 python=3 -c pytorch -c conda-forge -y
conda activate pasp_gnn
conda install pyg -c pyg -c conda-forge -y
`
when I run `from torch_geometric.nn.models.dimenet import SphericalBasisLayer`, it gives the same error: 
`OSError: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by .../lib/python3.9/site-packages/torch_spline_conv/_basis_cuda.so)`

But it will not happen if I install with Pytorch 1.9
Maybe a conflict problems between Pytorch 1.10 and torch_spline_conv?

**Comment by rusty1s** [comment]
What's your `GLIBC` version?
```
ldd --version
```

**Comment by yuhongyu0721** [comment]
> ldd --version

ldd (GNU libc) 2.17

**Comment by rusty1s** [comment]
I think the main problem is that our wheels are compiled with Ubuntu 18.04 which comes with GLIBC 2.27 be default. Any chance you can upgrade to a higher version? Otherwise, you may need to install the extensions via manual compiling:
```
pip install torch-spline-conv
```

**Comment by yuhongyu0721** [comment]
> I think the main problem is that our wheels are compiled with Ubuntu 18.04 which comes with GLIBC 2.27 by default. Any chance you can upgrade to a higher version? Otherwise, you may need to install the extensions via manual compiling:
> 
> ```
> pip install torch-spline-conv
> ```

Manual compiling pass. Thank you!

**Comment by chrisyeh96** [comment]
@rusty1s  - Since the wheels for `torch-spline-conv` links against GLIBC, and some systems have older versions of GLIBC installed, would it be possible to have `torch-spline-conv` (`pyg` conda version) link against a conda `glib` package (e.g., from the `anaconda` channel or `conda-forge` channel)? That way, versions would be managed within conda instead of relying on the native GLIBC installation which sometimes cannot be updated for various reasons.

**Comment by rusty1s** [comment]
Yes, I think this is possible. Am I correct to assume that all we need to do is install `glib` via `conda` prior to building the package [here](https://github.com/rusty1s/pytorch_spline_conv/blob/master/.github/workflows/building-conda.yml)?

**Comment by chrisyeh96** [comment]
> Yes, I think this is possible. Am I correct to assume that all we need to do is install `glib` via `conda` prior to building the package [here](https://github.com/rusty1s/pytorch_spline_conv/blob/master/.github/workflows/building-conda.yml)?

I wish I could help, but unfortunately I know very little about GitHub Actions and building packages. My original idea was that you could specify `glib` as a conda dependency of `pyg`.

**Comment by hnisonoff** [comment]
I am running into this issue and have been unable to resolve it by running `conda install -c conda-forge glib` before installing `pytorch-spline-conv`. Does anyone have suggestions for other solutions? Thanks!

**Comment by rusty1s** [comment]
For now, the only workaround is to install an older version of `torch-spline-conv` or avoid to install it in the first place (it's an optional dependency of PyG).

**Comment by yuchenlichuck** [comment]
I come across the same problems

**Comment by jiawei6636** [comment]
> I think the main problem is that our wheels are compiled with Ubuntu 18.04 which comes with GLIBC 2.27 by default. Any chance you can upgrade to a higher version? Otherwise, you may need to install the extensions via manual compiling:
> 
> ```
> pip install torch-spline-conv
> ```

How to install the extension via manual compiling?

**Comment by rusty1s** [comment]
`pip install` without the `-f` option will result in a manual compilation. It is similar to doing
```
git clone https://github.com/rusty1s/pytorch_spline_conv
cd pytorch_spline_conv
python setup.py install
```

**Comment by almumujin** [comment]
When I manually compiled torch-spline-conv：

r: invalid static_cast from type ‘const torch::OrderedDict<std::basic_string<char>, std::shared_ptr<torch::nn::Module> >’ to type ‘torch::OrderedDict<std::basic_string<char>, std::shared_ptr<torch::nn::Module> >&’
error: command '/usr/local/cuda-11.3/bin/nvcc' failed with exit code 1

**Comment by fengenxiang1123** [comment]
same error

OS : ubuntu 16.04
python : 3.9
pytorch: 1.10.2
CUDA 10.2
GCC version 5.4.0
PyG isntall : using "conda install pyg=2.0.4 -c pyg -c conda-forge" 
import torch_geometric successfully, but when I try to import torch_geometric.nn， this error occurred


I solved this by "pin uninstall torch-spline-conv" and "pip install torch-spline-conv -f https://data.pyg.org/whl/torch-${TORCH}+${CUDA}.html"
